### ISSUE TITLE
Criteria & initial steps for new repos

### ISSUE THREAD

**Issue by TPS** [rejuvenate/rejuvenate#1]
@ThomDietrich I just contributed
- https://github.com/rejuvenate/JSView
- https://github.com/rejuvenate/SortExceptions

They definitely have structural issues (e.g., XPIs should be in /Releases)… & I forgot to enable /Issues, /Wiki, &c _before_ transfer. 🤦

Most important question:  Should _all_ contributed repos have any kind of consistency upon acceptance here?

**Comment by Mottie** [comment]
Nice work everyone!

First off, I think we should continue our discussions here, or maybe in a Gitter channel, to keep the spamming of the isaacs/github channel down.

> How do we add repositories?

Adding repos is easy, but we should have some criteria (popularity, usefullness, etc) before saying a repo is ok to be forked.

> Who gets write access to repositories?

Write access should be granted to whomever asks for it, contributor access, not admin. Git makes it easy to revert problems so I think trusting people is better than restricting them.

Personally, there are a lot of languages I don't know, so I won't be doing much contributions. Maybe we should set up teams based on the repo language? Then invite people into the team when they request it.

>Do we/users with write access directly merge Pull Requests available in the official repository?
AND
How do we handle Issues and Pull Requests in these projects?

GitHub has set up a review process which I think you can enable to be always required. That should allow team members to review and merge PRs as needed.

> Do we actively promote the new fork?

I think for now we could open an issue in the original repo with a boilerplate message saying we have forked the repo in order to keep it alive and that the original author is welcome to join in or whatever. We can work on actual promotion later.

**Comment by ThomDietrich** [comment]
Wonderful @Mottie, I think we are mostly on the same page here ;)

Please check what I've already written on https://github.com/rejuvenate/rejuvenate/blob/master/README.md

> How do we add repositories?

I wouldn't add random projects but trust in contributors. The vision of this organization (Let's call it community from now on) should be to create and maintain forks of abandoned but still relevant projects, **still receiving unanswered Issues and Pull Requests**. These projects would I believe fulfill your criteria (popularity, usefulness).

Regarding the How: If a repository matches this profile, I believe it is safe to create a fork and start working in it. A few settings need to be changed (e.g. activate Issues).

> Who gets write access to repositories?

I agree with the following: "Write access should be granted to whomever asks for it, contributor access, not admin.". The whole point of the Rejuvenate community should be to allow the "community" to contribute freely.
I would additionally try to motivate as many users as possible to join the organization, for the purpose of taking part in review processes (see below).

>> Personally, there are a lot of languages I don't know, so I won't be doing much contributions. Maybe we should set up teams based on the repo language? Then invite people into the team when they request it.

That's a great idea... for later maybe? I wouldn't over-complicate it for now. I believe the task of organization members is mostly to evaluate the value and quality of PRs and react on comments. Not each and every one will need to be a $LANGUAGE expert. Let's see how it goes!? ;)

> Do we/users with write access directly merge Pull Requests available in the official repository?
> How do we handle Issues and Pull Requests in these projects?

>> GitHub has set up a review process

Exactly what I was getting at :smiley: :+1: I believe we should define a guideline. Could be as easy as "Each PR needs at least two Approvals to be Accepted".

(In another project I'm involved in we use Approvals to validate multiple aspects, mainly "Functionality & Code Quality" and "Documentation & Code Style". Maybe something like this can be arranged at a later point.)

> Do we actively promote the new fork?

>> I think for now we could open an issue in the original repo with a boilerplate message saying we have forked the repo in order to keep it alive and that the original author is welcome to join in or whatever. We can work on actual promotion later.

Agreed. Be polite and inform the original developer. Make users aware at the same time. No aggressive advertisement ;)

----

I have two followup questions:

> Is a user with repository write access allowed to write directly to the master branch or does he have to create PRs with the need for approval just as everyone else?

My opinion here would be that this should be allowed but PR approval should be the preferred choice.

> Many projects we are going to fork will contain unanswered PRs. Screening and merging these should be of high priority. How do we go about that?

The ownership of commits should not be altered or removed (at least not without the consent of the author). That's why I would suggest to create a new Pull Request (Merge or Rebase) for these changes in the community repository, mentioning the author as mandatory reviewer.

----

Both of you should now have write access to this Repository. PLEASE add everything we agree on to the README file. :wink:

I'm excited to get this project going!

**Comment by TPS** [comment]
@rejuvenate/community Feedback on edge case, then, please: I _transferred_ (not forked) 2 repos over, because they're not _my_ work, but 2 GPL'ed (abandoned) Firefox extensions for which the actual authors seem not to exist online _anyplace_, much less GitHub.  Should have I requested to fork,  instead, in this kinda case?

**Comment by Mottie** [comment]
Maybe we should set up another organization for repositories that need support but don't have someone supporting it yet - "rejuvenate-que", or something.

Once someone is available to support it, we can then move it into this main organization and set up or assign a team.

**Comment by TPS** [comment]
@rejuvenate/community  That's good, but, for instance, in the case of the XPI (& other script-based, &c) repos, an intermediate possibility exists, too: No full maintainer, but, if someone contributes a PR, it's pretty simple to whip up an unofficial (non-signed) XPI, collect :shipit:/WFMs/👍s, & then vote it a proper release. (That's the reason I'd them setup as PR-only when I kept them.)

Also, I'm concerned a separate org like that _may_ dilute attention & just collect abandonware in 1 spot w/o anyone noticing.…

**Comment by ThomDietrich** [comment]
Hey guys, I don't think we have to be too picky which projects find their way here. If there is no activity in a repo it will move out of sight and if one is indeed a problem for the organization, we could still remove it.
The only concern I have personally is of legal matter. I think we should only accept projects that have a correctly applied open source software license.

> repositories that need support

Is that a problem? First and foremost I would say that will always be the person bringing the repo into the organization. Secondly I don't think that this person is strictly needed as the review process will be distributed throughout the organization. I would not over-complicate things, especially now in the beginning.

I'll extend the README over the weekend. Related to that, I'd still be interested in your opinions about the two followup questions.

**Comment by TPS** [comment]
Yes, I absolutely agree: Along w/ #3, each repo should be _required_ to provide/prove adequate licensing (& perhaps go thru http://www.addalicense.com/, if allowed for forks?) before acceptance.


F↑Q#1:
PRs all the way…

F↑Q#2:
… but maybe each sub-community should decide if they want voting, benevolent dictatorship, or whatever re:  how PRs are handled (accepted, declined, ruminated, &c) just as long all action/inaction is justified (no complete ignoring w/o explanation)!  


Given correct licensing, forking even these Rejuvenate repos shouldn't cause problems, ideally, if someone's miffed their PR wasn't accepted, right?

However, given the nature of the repos that we wish to attract, wouldn't giving the original code author _mandatory_ review privilege cause the same problem we're trying to solve?  If said author was responsive, there wouldn't be need to fork _here_!

**Comment by TPS** [comment]
@rejuvenate/community  I renamed this issue to better reflect what we're discussing.  Please feel free to further clarify.

Also, https://lists.mayfirst.org/pipermail/guardian-dev/2017-February/005180.html came through my inbox today.  While I believe it'd be inappropriate to actually step up as maintainers, it'd _probably_ be appropriate to fork, but what else?

**Comment by ThomDietrich** [comment]
Hey guys, just a short update from my side. I was a bit swamped the last couple of days/weeks and didn't find the time to continue here. Hopefully soon! @Mottie what's your opinion on the two questions above?

Best! Thomas

**Comment by Mottie** [comment]
I agree on requiring an open-source license before allowing a fork to be added. And as @TPS stated, we can minimize the review and accepting of PRs until we establish a sub- community for that repo and let them decide.

**Comment by TPS** [comment]
@rejuvenate/community I figured I should RFC https://github.com/github/open-source-survey (which popped up while I was in a minimally-active project here on GitHub) vs https://github.com/github/balanced-employee-ip-agreement (🎩/T @ThomDietrich).  Each of these made me think about some points we mentioned either in this issue or #3.  Whenever y'all get a chance, tell me what you think.
